### ISSUE TITLE
[Bug] Model mismatches when running gen_ViT_MoE_weight.py

### ISSUE THREAD

**Issue by xyhe1996** [Adlith/MoE-Jetpack#4]
### 分支

main 分支 (mmpretrain 版本)

### 描述该错误

When I performed debugging with the command `python moejet/tools/gen_ViT_MoE_weight.py` as mentioned in README file, I noted that the default model to initialize moe weights was `vit_tiny`. 
However, there are some codes in lines [885](https://github.com/Adlith/MoE-Jetpack/blob/45214a6742e6242fddc07683a8a8ef8f6cb6b10a/moejet/tools/gen_ViT_MoE_weight.py#L885), [891](https://github.com/Adlith/MoE-Jetpack/blob/45214a6742e6242fddc07683a8a8ef8f6cb6b10a/moejet/tools/gen_ViT_MoE_weight.py#L891), [945](https://github.com/Adlith/MoE-Jetpack/blob/45214a6742e6242fddc07683a8a8ef8f6cb6b10a/moejet/tools/gen_ViT_MoE_weight.py#L945), [951](https://github.com/Adlith/MoE-Jetpack/blob/45214a6742e6242fddc07683a8a8ef8f6cb6b10a/moejet/tools/gen_ViT_MoE_weight.py#L951), [952](https://github.com/Adlith/MoE-Jetpack/blob/45214a6742e6242fddc07683a8a8ef8f6cb6b10a/moejet/tools/gen_ViT_MoE_weight.py#L952) and [957](https://github.com/Adlith/MoE-Jetpack/blob/45214a6742e6242fddc07683a8a8ef8f6cb6b10a/moejet/tools/gen_ViT_MoE_weight.py#L957) for `vit_s` and the name for the saved checkpoint also includes `vit_t` and `vit_s`.
It makes me confused. Should `vit_s` in the above lines be corrected to `vit_t` as performing moe weights initialization for  `vit_t`? If I miss anything, please let me know. THKS    

### 环境信息

NA

**Comment by ConanMusk** [comment]
你也在使用这个仓库吗？可否交流 VX 18770563607

**Comment by Adlith** [comment]
Hi, thanks for your feedback!

This is not a bug. The current behavior is expected: the MoE weight initialization script can initialize ViT-Tiny's MoE weights using weights extracted from ViT-Small, or alternatively, initialize directly from a ViT-Tiny checkpoint.

Both paths are supported intentionally:

- vit_s is used for recycling knowledge from a larger model.

- vit_tiny is for direct initialization from a same-size backbone.

Hope this clears up the confusion. Let us know if anything is still unclear!

**Comment by xyhe1996** [comment]
> Hi, thanks for your feedback!
> 
> This is not a bug. The current behavior is expected: the MoE weight initialization script can initialize ViT-Tiny's MoE weights using weights extracted from ViT-Small, or alternatively, initialize directly from a ViT-Tiny checkpoint.
> 
> Both paths are supported intentionally:
> 
> * vit_s is used for recycling knowledge from a larger model.
> * vit_tiny is for direct initialization from a same-size backbone.
> 
> Hope this clears up the confusion. Let us know if anything is still unclear!

Thank you for your reply. I have two additional questions:
1. During the weight generation process, the val_loader uses 38,400 samples extracted from the training set in [gen_ViT_MoE_weight.py#L907](https://github.com/Adlith/MoE-Jetpack/blob/45214a6742e6242fddc07683a8a8ef8f6cb6b10a/moejet/tools/gen_ViT_MoE_weight.py#L907). Could you further explain the reason for it, or did I miss something?
2. When I used the MoE Jetpack Weights [(epoch_300.pth)](https://drive.google.com/file/d/1em88nNFf6vVNA4ert4URlSQMaTwMxPKY/view?usp=sharing) you provided and ran the following command:
`CUDA_VISIBLE_DEVICES=0 PORT=29500 ./tools/dist_test.sh moejet/configs/timm/vit_tiny_dual_moe_timm_21k_ft.py './epoch_300.pth' 1` for validation on ImageNet, the top-1 accuracy was 80.358. Does this result correspond to the 79.9 (+2.8) shown in your Table 1(a)?

**Comment by Adlith** [comment]
Thank you for your questions!
1. Regarding the use of 38,400 samples in val_loader during weight generation:  
This subset is used to estimate the importance of hidden neurons in the pretrained dense model. Specifically, we pass a small batch of ImageNet-1K images through the pretrained network to record neuron activations in FFN layers. These activations are then used to calculate each neuron's importance, guiding the selection of neurons for initializing MoE experts. This follows the importance-based weight sampling strategy described in Section 3.1 of the paper.

2. Regarding the top-1 accuracy of 80.358:  
Yes, this result is expected. After the paper submission, we refactored and optimized the codebase, which led to improved performance compared to the originally reported number.

**Comment by xyhe1996** [comment]
appreciate your reply, thanks again~
